core-api/src/main/java/com/optimizely/ab/Optimizely.java

```diff
@@ -124,10 +125,15 @@ private Optimizely(@Nonnull EventHandler eventHandler,
 
         if (odpManager != null) {
             odpManager.getEventManager().start();
-            if (getProjectConfig() != null) {
+            if (!(projectConfigManager instanceof PollingProjectConfigManager)) {
```

Comment:
PollingProjectConfigManager can also have a preset projectConfig from a fallback datafile. In that case, we have to call `updateODPSettings` for the projectConfig immediately (we may not get notification later if the fallback is updated already).
I think we can keep the `getProjectConfig() != null` condition as before, with the change that `getProjectConfig() returns null immediately if config is not available.

Reply:
`getProjectConfig() != null`  condition is not possible without wait, with our current implementation. However, we can use the similar approach as we are using in Csharp, adding additional function of `getCachedConfig()` in `ProjectConfigManager`. Now the only concern I have, related to this approach, is that do we want to enforce its implementation or return null by default as we are doing for `getSDKKey()` function? 
In my opinion we should enforce it because if anyone uses custom `ProjectConfigManager` without giving `getCachedConfig` implementation then they won't be able use ODP as it only gets config using this new function.

Reply:
@mnoman09 Agreed. Let's make them (`getSdkKey` and `getCachedConfig`) mandatory changes and keep them in the list of breaking changes for release.